django/http/request.py

```diff
@@ -19,11 +20,17 @@
 from django.utils.deprecation import RemovedInDjango40Warning
 from django.utils.encoding import escape_uri_path, iri_to_uri
 from django.utils.functional import cached_property
-from django.utils.http import is_same_domain, limited_parse_qsl
+from django.utils.http import is_same_domain
 from django.utils.regex_helper import _lazy_re_compile
 
 from .multipartparser import parse_header
 
+if (3, 6, 8) <= sys.version_info[:3] < (3, 7, 0) or (3, 7, 2) <= sys.version_info[:3]:
```

Comment:
I'm not sure about this check. I would prefer to use `urllib.parse.parse_qsl` only on Python 3.8 which seems reasonable because you backported version from Python 3.8 to the `django.utils.http.parse_qsl`.

Alternatively, we can check `urllib.parse.parse_qsl` signature rather then Python version on the grounds that adding `max_num_fields` was a security patch so older versions may have this patch:
```
Python 3.6.3
[GCC 7.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> from urllib.parse import parse_qsl
>>> help(parse_qsl)
​
Help on function parse_qsl in module urllib.parse:
​
parse_qsl(qs, keep_blank_values=False, strict_parsing=False, encoding='utf-8', errors='replace', max_num_fields=None)
    Parse a query given as a string argument.
```
but I would prefer to use `urllib.parse.parse_qsl` only on Python 3.8.